test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +103,51 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const readable = streamifyArray([ 1, 2, 3 ]);
+
+    source.getRepresentation = jest.fn(async(): Promise<any> =>
+      new Promise((resolve): any => delayedResolve(resolve, 'getRepresentation', { data: readable } as
+        Representation)));
```

Comment:
I would put this in the `beforeEach` call and replace the current mock of `getRepresentation` in there. This way you don't have to expose the `delayedResolve` above as well. Maybe the original GET this will fail then but you should remove that one anyway.